force-app/main/default/classes/SOQL.cls

```diff
@@ -288,20 +334,36 @@ public with sharing class SOQL {
      */
     public static List<Account> findContactsByLeadEmailDomains(Set<Id> leadIds) {
         // Query the Leads
+        List<Lead> leadList = [SELECT Id,Email, Company FROM Lead WHERE Id IN :leadIds];
 
         // Create a set of email domains from the Leads
+        Set<String> emailSet = new Set<String>();
 
         // Loop through each Lead
-
+        for(Lead eachLead : leadList){
             // Get the email from the Lead
-
             // Check if the email is not null
-
-                // Add the email domain to the set
-
-        
+            if(eachLead.Email != null){
+                 // Add the email domain to the set
+                 emailSet.add(eachLead.Email.substringAfter('@'));
+                 System.debug(emailSet);
+            }
+        }             
         // Query the Accounts
-        return null; // Replace null with the variable you used to store the result
+        List<Account> finalAccList = new List<Account>();
+        List<Account> accList= [SELECT Id, Name,Website FROM Account WHERE Website != null];
+            for(Account eachAcc : accList){
+                for(String domain : emailSet){
+                    System.debug(eachAcc);
+                    System.debug(domain);
+                    if(eachAcc.Website.contains(domain) ){
+                        finalAccList.add(eachAcc);
+                    }
+                }
+            }
```

Comment:
This is a great first pass and how I would write it initially. 
Is there an optimization here that can be done?
Is there a way to query for only the accounts that match the emailSet (domains) that we gather from the leads?